docsource/installation.rst

```diff
@@ -0,0 +1,207 @@
+Installation Instructions
+=========================
+
+Local Deployment
+-----------------
+Follow these instructions for a local deployment once the :ref:`System Requirements` have been satisfied. These instructions are known to work on Ubuntu 16.04 LTS. The rest of this playbook assumes that you have all dependency binaries in your system environment path variable. At a high level, the steps are -
+
+.. code-block:: text
+
+    1. Start ElasticSearch
+    2. Start Kibana
+    3. Configure Workspace listeners to write events to Search mongo db
+    4. Restart Workspace service
+    5. Create a Workspace data type
+    6. Configure KBaseSearchEngine
+    7. Start worker
+    8. Start coordinator
+    9. Verify ElasticSearch index
+
+1. Open a new terminal and start ElasticSearch.
+
+.. note::
+
+    Elastic Search can only be started up by a non-root user
+
+.. code-block:: bash
+
+    $ elasticsearch
+
+2. Open a new terminal and start Kibana.
+
+.. code-block:: bash
+
+   $ kibana
+
+3. Configure the Workspace listeners to write events to the Search mongodb.
+
+.. code-block:: bash
+
+    $ gedit [PATH_TO_YOUR_WORKSPACE_DIR]/deploy.cfg
+
+Make the following changes -
+
+.. code-block:: cfg
+
+    listeners = Search
+    listener-Search-class = us.kbase.workspace.modules.SearchPrototypeEventHandlerFactory
+    listener-Search-config-mongohost = localhost
+    listener-Search-config-mongodatabase = Search_test
+    listener-Search-config-mongouser = ""
+    listener-Search-config-mongopwd = ""
+
+4. Restart the Workspace Service. (See section on `Deploying the Workspace Service locally <https://github.com/kbase/workspace_deluxe/blob/dev-candidate/docsource/developers.rst>`_)
+
+5. Open a new terminal and save the following document as Empty.spec. Then load into ipython and save the object to the Workspace. Saving a new object will cause the Workspace listener to write a new event to the mongo instance.
+
+.. code-block:: javascript
+
+    module Empty {
+
+        /* @optional foo */
+        typedef structure {
+            int foo;
+        } AType;
+    };
+
+.. code-block:: bash
+
+    $ ipython
+
+    In [1]: spec = open("[PATH_TO_SPEC]/Empty.spec").read()
+    In [2]: ws.request_module_ownership('Empty')
+    In [3]: ws.administer({'command': 'listModRequests'})
+    Out[4]:
+    [{u'moduleName': u'Empty', ...}]
+    In [5]: ws.administer({'command': 'approveModRequest', 'module': 'Empty'})
+    In [6]: ws.register_typespec({'spec': spec, 'new_types': ['AType'], 'dryrun': 0})
+    Out[7]: {u'Empty.Atype-0.1': ....}
+    In [8]: ws.register_module('Empty')
```

Comment:
I think you mean release_module()
https://ci.kbase.us/services/ws/docs/Workspace.html#funcdefWorkspace.release_module

Reply:
Fixed.